xml/api-result-0.1.rng

```diff
@@ -0,0 +1,124 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<grammar xmlns="http://relaxng.org/ns/structure/1.0"
+         datatypeLibrary="http://www.w3.org/2001/XMLSchema-datatypes">
+
+    <start>
+        <ref name="element-pacemaker-result"/>
+    </start>
+
+    <define name="element-pacemaker-result">
+        <element name="pacemaker_result">
+            <attribute name="api_version"> <text /> </attribute>
+            <attribute name="request"> <text /> </attribute>
+            <optional>
+                <choice>
+                    <element name="installed_fence_devices">
+                        <attribute name="count"> <data type="integer" /> </attribute>
+                        <ref name="device-list" />
+                    </element>
+                    <element name="registered_fence_devices">
+                        <attribute name="count"> <data type="integer" /> </attribute>
+                        <ref name="device-list" />
+                    </element>
+                    <element name="fence_targets">
+                        <ref name="target-list" />
+                    </element>
+                    <element name="metadata"> <text /> </element>
+                    <element name="fencing_history">
+                        <attribute name="count"> <data type="integer" /> </attribute>
+                        <optional>
+                            <choice>
+                                <ref name="element-last-fenced" />
+                                <ref name="fencing-history-list" />
+                            </choice>
+                        </optional>
+                    </element>
+                    <ref name="element-validation" />
+                </choice>
+            </optional>
+            <element name="status">
+                <attribute name="code"> <data type="integer" /> </attribute>
+                <text />
+            </element>
+        </element>
+    </define>
+
+    <define name="device-list">
+        <zeroOrMore>
+            <ref name="device" />
+        </zeroOrMore>
+    </define>
+
+    <define name="device">
+        <element name="device"> <data type="ID" /> </element>
```

Comment:
Again, may have missed something from the overly long discussion,
but I'd be rather careful about `ID`, since it incurs problems related
to sharing single namespace for multiple purposes.  Is it meant here to
exclude redundancy amongst multiple installed/registered fence devices,
so that each unique one can be member at most in a single domain only?

Reply:
Ok, I see that currently, there's a simplistic *targeted* request-response
approach, which would preclude both sets being stated at the same time,
but let's think about possibly making compositions in the future.
Is it a legitimate request to ask for both of these sets at once, so
that the UI can work with both without extra roundtrips?

Reply:
Managed to dig this out (longish reviews with multiple iteration really do
suck per how they are presented on GitHub):
https://github.com/ClusterLabs/pacemaker/pull/1707#discussion_r264879864
but I don't follow that justification.  CIB indeed uses the IDs, and for
two purposes:

* enforcing uniqueness (although in an single-namespace-only afflicted way)

* enforcing referential integrity (ditto, but better than nothing)

I am missing the purpose it'd be used for here.

If there's none per se, yet lexical compatibility with `ID` is requested,
just use `NCName`.

[Note that I am starting to thing that using multiple namespaces is a way
to split the shared namespace of uniqueness into disjoint spaces, which
might be very desirable.]

Reply:
I agree that NCName (i.e. this must follow the naming limitations of an ID) was what we actually had in mind here, rather than uniqueness within the output.